### PR TITLE
chore(policy): remove architecture constraints validation from validator

### PR DESCRIPTION
**Pull request by nvinuesa**
This patch removes the architectures validation from the constraint validation, namely it now stops retrieving the cloud image metadata supported architectures and the manually provided machines architectures.
This last part was added to fix a bug (https://bugs.launchpad.net/juju/+bug/1946639), but we now prefer to fail at provision-time when the user passes an unsupported architecture.

## Checklist

- [X] Code style: imports ordered, good names, simple structure, etc
- [ ] ~Comments saying why design decisions were made~
- [ ] ~Go unit tests, with comments saying what you're testing~
- [ ] ~[Integration tests](https://github.com/juju/juju/tree/main/tests), with comments saying what you're testing~
- [ ] ~[doc.go](https://discourse.charmhub.io/t/readme-in-packages/451) added or updated in changed packages~

## QA steps

N/A.

## Links

**Jira card:** JUJU-6368

### CODE REVIEW

**Review by manadart — COMMENTED** [review]
Plenty more of state that we can throw away here.

File: state/policy.go [hunk 1]
```diff
@@ -60,40 +58,6 @@ func (st *State) constraintsValidator() (constraints.Validator, error) {
...
-				Region: region,
-			},
-		)
-		if err != nil {
-			return nil, errors.Annotate(err, "querying supported architectures")
-		}
-
-		// Also include any arches that belong to manually provisioned machines
-		// See LP1946639.
-		manualMachArches, err := st.GetManualMachineArches()
```
[the remark this answers is not in the document]
Comment (nvinuesa) [reply state/policy.go:87]: I was going to clean up everything (included this function) when finishing the machine cloud instance epic, but it's OK to remove now. See bd1e949c27.

File: state/policy.go [hunk 2]
```diff
@@ -60,40 +58,6 @@ func (st *State) constraintsValidator() (constraints.Validator, error) {
...
-	model, err := st.Model()
-	if err != nil {
-		return nil, errors.Annotate(err, "getting model")
-	}
-	if region := model.CloudRegion(); region != "" {
-		cfg, err := model.ModelConfig(stdcontext.Background())
-		if err != nil {
-			return nil, errors.Trace(err)
-		}
-		arches, err := st.CloudImageMetadataStorage.SupportedArchitectures(
```
[the remark this answers is not in the document]
Comment (nvinuesa) [reply state/policy.go:75]: Done, see bd1e949c27.

**Review by SimonRichardson — APPROVED** [review]

**Pull request merged**